### PR TITLE
test(storagetransfer): Isolate STS Test Credentials

### PR DESCRIPTION
**Pull request by danielbankhead**
Overview:
- Removes shared secret file in favor of secret manager (the secret manager secret now includes the relevant env variables)
- Utilizes `STS_AWS_SECRET_NAME` env variable for STS tests

### CODE REVIEW

File: .kokoro/tests/run_tests.sh [hunk 1]
```diff
@@ -96,12 +96,6 @@ fi
 source ./testing/test-env.sh
 export GOOGLE_APPLICATION_CREDENTIALS=$(pwd)/testing/service-account.json
 
-# Import secrets for AWS integration testing. This can be used for products
```
Comment (danielbankhead) [inline .kokoro/tests/run_tests.sh:99]: We want to use secret store instead.

Undoes this change in this commit: https://github.com/GoogleCloudPlatform/python-docs-samples/pull/7091/files#diff-ff2ac0c2d9eac8a4fc58f24e20895504b9c25a1b6d6c1f6f55a5978c1f5aefc3

File: storagetransfer/conftest.py [hunk 2]
```diff
@@ -32,30 +32,76 @@ def project_id():
...
+
+    # retrieve from secret manager
+    response = client.access_secret_version(request={"name": name})
+
+    # parse secret from secret manager
+    secret = json.loads(response.payload.data.decode("UTF-8"))
+
+    aws_secret_manager_cache = {
+        'aws_access_key_id': secret['AccessKeyId'],
+        'aws_secret_access_key': secret['SecretAccessKey']
```
Comment (dandhlee) [inline storagetransfer/conftest.py:61]: nit: The last item in a list of items (in `list`, `dict`, `set`, etc) should end with a comma.
```suggestion
        'aws_secret_access_key': secret['SecretAccessKey'],
```

File: storagetransfer/conftest.py [hunk 3]
```diff
@@ -32,30 +32,76 @@ def project_id():
...
+    aws_access_key_id = os.environ.get("AWS_ACCESS_KEY_ID")
+    aws_secret_access_key = os.environ.get("AWS_SECRET_ACCESS_KEY")
+
+    if sts_aws_secret_name:
+        return aws_retrieve_from_secret_manager(sts_aws_secret_name)
+    else:
+        return {
+            'aws_access_key_id': aws_access_key_id,
+            'aws_secret_access_key': aws_secret_access_key
+        }
```
Comment (dandhlee) [inline storagetransfer/conftest.py:78]: `else` in this case is optional, so not having it reduces nesting and still remains straightforward code.
```suggestion
    
    return {
        'aws_access_key_id': aws_access_key_id,
        'aws_secret_access_key': aws_secret_access_key
    }
```

File: storagetransfer/conftest.py [hunk 4]
```diff
@@ -32,30 +32,76 @@ def project_id():
...
+
+
+@ pytest.fixture(scope='module')
```
Comment (dandhlee) [inline storagetransfer/conftest.py:81]: Any reason for the extra space here? I couldn't find whether this is new or not.

**Review by dandhlee — CHANGES_REQUESTED** [review]
The secret version ID seems to come in (`projects/*/secrets/*/version/*`) instead of the expected (`projects/*/secrets/*/versions/*`) format, missing `s` on `version` from the tests.

File: storagetransfer/conftest.py (hunk 4 above)
Comment (danielbankhead) [reply storagetransfer/conftest.py:81]: Yea, it's a little strange, I'll revert this auto-formatting.

File: storagetransfer/conftest.py (hunk 3 above)
Comment (danielbankhead) [reply storagetransfer/conftest.py:78]: Thanks, fixing

**Review by dandhlee — APPROVED** [review]
LGTM, all tests for `storagetransfer` are passing. Failing tests are from `.kokoro/tests/run_tests.sh` changes and can be ignored for this PR.

**Review by leahecole — APPROVED** [review]

**Review by cojenco — APPROVED** [review]

**Comment by dandhlee** [comment]
Had previous tests passing, admin merging (some failures are expected, but they should be unrelated to storagetransfer)

**Pull request merged**